client/src/app/openslides-main-module/services/app-load.service.ts

```diff
@@ -54,12 +55,19 @@ const appConfigs: AppConfig[] = [
     providedIn: 'root'
 })
 export class AppLoadService {
+    private modelMapper: CollectionMapperService;
+    private mainMenuService: MainMenuService;
+    private fallbackRoutesService: FallbackRoutesService;
+
     public constructor(
         private injector: Injector,
-        private modelMapper: CollectionMapperService,
-        private mainMenuService: MainMenuService,
-        private fallbackRoutesService: FallbackRoutesService
-    ) {}
+    ) {
+        // Circumventing the standard CI here to set the injector before any other services are loaded
+        AppInjector.setInjector(injector);
```

Comment:
Hmm, this seems a little bit hacky, doesn't it? Why don't you set the injector as in your [example](https://devblogs.microsoft.com/premier-developer/angular-how-to-simplify-components-with-typescript-inheritance/)?

```suggestion
// in main.ts
platformBrowserDynamic()
    .bootstrapModule(OpenSlidesMainModule)
    .then(moduleRef => AppInjector.setInjector(moduleRef.injector))
    .catch(err => console.error(err));
```

Reply:
Because this way, angular loads some components before the bootstrap promise resolves, which fails because the injector is not present yet. This was the best place I found to set the injector before anything else is loaded.